### ISSUE TITLE
Command Line Tools Console not working

### ISSUE THREAD

**Issue by xgathos** [VladRassokhin/intellij-hcl#123]
Hi

I can not use the command line tools in PhpStorm 2017.2.4 with 0.6.9 and 0.6.9.1 -> it is not possible to press enter.

I found this exception in idea.log:

```log
2017-11-19 11:38:12,508 [  20837]  ERROR - llij.ide.plugins.PluginManager - Macro 'TerraformExecPath' should be registered in PathMacros! 
java.lang.AssertionError: Macro 'TerraformExecPath' should be registered in PathMacros!
	at com.intellij.ide.macro.MacroManager.b(MacroManager.java:109)
	at com.intellij.ide.macro.MacroManager.<init>(MacroManager.java:104)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:103)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:80)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:63)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.getComponentInstance(ServiceManagerImpl.java:220)
	at com.intellij.util.pico.DefaultPicoContainer.getLocalInstance(DefaultPicoContainer.java:239)
	at com.intellij.util.pico.DefaultPicoContainer.getComponentInstance(DefaultPicoContainer.java:206)
	at com.intellij.openapi.components.ServiceManager.doGetService(ServiceManager.java:48)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:38)
	at com.intellij.ide.macro.MacroManager.getInstance(MacroManager.java:46)
	at com.jetbrains.php.framework.FrameworkUtils.resolveMacros(FrameworkUtils.java:197)
	at com.jetbrains.php.framework.FrameworkEditorActionHandler$UserInputAnalyzer.<init>(FrameworkEditorActionHandler.java:444)
	at com.jetbrains.php.framework.FrameworkEditorActionHandler.executeWriteAction(FrameworkEditorActionHandler.java:113)
	at com.intellij.openapi.editor.actionSystem.EditorWriteActionHandler$1.run(EditorWriteActionHandler.java:50)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1030)
	at com.intellij.openapi.editor.actionSystem.EditorWriteActionHandler.doExecute(EditorWriteActionHandler.java:63)
	at com.intellij.openapi.editor.actionSystem.EditorActionHandler$4.perform(EditorActionHandler.java:217)
	at com.intellij.openapi.editor.actionSystem.EditorActionHandler.doIfEnabled(EditorActionHandler.java:109)
	at com.intellij.openapi.editor.actionSystem.EditorActionHandler.execute(EditorActionHandler.java:214)
	at com.intellij.openapi.editor.actionSystem.EditorAction.lambda$actionPerformed$0(EditorAction.java:98)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.a(CoreCommandProcessor.java:149)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:109)
	at com.intellij.openapi.editor.actionSystem.EditorAction.actionPerformed(EditorAction.java:107)
	at com.intellij.openapi.editor.actionSystem.EditorAction.actionPerformed(EditorAction.java:86)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:216)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:233)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$1.performAction(IdeKeyEventDispatcher.java:576)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.a(IdeKeyEventDispatcher.java:625)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:193)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:624)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.a(IdeKeyEventDispatcher.java:479)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:213)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:621)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
2017-11-19 11:38:12,509 [  20838]  ERROR - llij.ide.plugins.PluginManager - PhpStorm 2017.2.4  Build #PS-172.4155.41 
2017-11-19 11:38:12,509 [  20838]  ERROR - llij.ide.plugins.PluginManager - JDK: 1.8.0_152-release 
2017-11-19 11:38:12,509 [  20838]  ERROR - llij.ide.plugins.PluginManager - VM: OpenJDK 64-Bit Server VM 
2017-11-19 11:38:12,509 [  20838]  ERROR - llij.ide.plugins.PluginManager - Vendor: JetBrains s.r.o 
2017-11-19 11:38:12,509 [  20838]  ERROR - llij.ide.plugins.PluginManager - OS: Windows 10 
2017-11-19 11:38:12,509 [  20838]  ERROR - llij.ide.plugins.PluginManager - Last Action: EditorEnter 
```

Can anybody reproduce this? As soon as I uninstall the plugin, the command line tools window is working again.

Thank you.
André

**Comment by VladRassokhin** [comment]
Thanks you for reporting. Turns out it's [impossible to register custom macros](https://github.com/JetBrains/intellij-community/commit/fdf6dbc#diff-bac8f371a98d4d44636fa8e17aaceca4L109) from plugin in IntelliJ platform older than 2017.3.
Will try to fix/workaround that ASAP.

**Comment by VladRassokhin** [comment]
I've fixed problem and uploaded new version (0.6.9.2) into plugins repository, please try it

**Comment by xgathos** [comment]
Thank you, it's working!

Thank you so much for the fix and for this plugin! :)
